src/vs/workbench/contrib/positronConnections/browser/components/schemaNavigation.tsx

```diff
@@ -0,0 +1,301 @@
+ *  Licensed under the Elastic License 2.0. See LICENSE.txt for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import React, { useEffect, useRef, useState, MouseEvent } from 'react';
+import { useStateRef } from 'vs/base/browser/ui/react/useStateRef';
+import { DisposableStore } from 'vs/base/common/lifecycle';
+import { usePositronConnectionsContext } from 'vs/workbench/contrib/positronConnections/browser/positronConnectionsContext';
+import * as DOM from 'vs/base/browser/dom';
+import { IPositronConnectionEntry } from 'vs/workbench/services/positronConnections/browser/positronConnectionsUtils';
+import { ActionBar, ACTION_BAR_HEIGHT } from 'vs/workbench/contrib/positronConnections/browser/components/schemaNavigationActionBar';
+import { FixedSizeList as List } from 'react-window';
+import { positronClassNames } from 'vs/base/common/positronUtilities';
+import 'vs/css!./schemaNavigation';
+import { ViewsProps } from 'vs/workbench/contrib/positronConnections/browser/positronConnections';
+
+export interface SchemaNavigationProps extends ViewsProps { }
+
+export const SchemaNavigation = (props: React.PropsWithChildren<SchemaNavigationProps>) => {
+
+	const context = usePositronConnectionsContext();
+	const { height, activeInstanceId, setActiveInstanceId } = props;
+
+	// We're required to save the scroll state because browsers will automatically
+	// scrollTop when an object becomes visible again.
+	const [, setScrollState, scrollStateRef] = useStateRef<number[] | undefined>(undefined);
+	const innerRef = useRef<HTMLElement>(undefined!);
+	useEffect(() => {
+		const disposableStore = new DisposableStore();
+		disposableStore.add(context.reactComponentContainer.onSaveScrollPosition(() => {
+			if (innerRef.current) {
+				setScrollState(DOM.saveParentsScrollTop(innerRef.current));
+			}
+		}));
+		disposableStore.add(context.reactComponentContainer.onRestoreScrollPosition(() => {
+			if (scrollStateRef.current) {
+				if (innerRef.current) {
+					DOM.restoreParentsScrollTop(innerRef.current, scrollStateRef.current);
+				}
+				setScrollState(undefined);
+			}
+		}));
+		return () => disposableStore.dispose();
+	}, [context.reactComponentContainer, scrollStateRef, setScrollState]);
+
+	const [selectedId, setSelectedId] = useState<string>();
+	const activeInstance = context.connectionsService.getConnections().find(item => item.id === activeInstanceId);
+
+	const [entries, setEntries] = useState<IPositronConnectionEntry[]>(activeInstance?.getEntries() || []);
+
+	useEffect(() => {
+		if (!activeInstance) {
+			return;
+		}
+		const disposableStore = new DisposableStore();
+
+		disposableStore.add(activeInstance.onDidChangeEntries((entries) => {
+			setEntries(entries);
+		}));
+
+		disposableStore.add(activeInstance.onDidChangeStatus((active) => {
+			if (!active) {
+				setActiveInstanceId(undefined);
+			}
+		}));
+
+		activeInstance.refreshEntries();
```

Comment:
`activeInstance.refreshEntries()` is not awaited / error handled.

Reply:
Resolved in 6ebfa7f5ed4ca3244d728933dc99a42b9b4f9879